reframe/core/schedulers/sge.py

```diff
@@ -0,0 +1,162 @@
+        )
+
+    def _emit_lselect_option(self, job):
+        num_tasks_per_node = job.num_tasks_per_node or 1
+        num_cpus_per_task = job.num_cpus_per_task or 1
+        num_nodes = job.num_tasks // num_tasks_per_node
+        num_cpus_per_node = num_tasks_per_node * num_cpus_per_task
+        select_opt = ''
+        self.TASKS_OPT.format(
+            num_nodes=num_nodes,
+            num_tasks_per_node=num_tasks_per_node,
+            num_cpus_per_node=num_cpus_per_node
+        )
+
+        # Options starting with `-` are emitted in separate lines
+        rem_opts = []
+        verb_opts = []
+        for opt in (*job.sched_access, *job.options, *job.cli_options):
+            if opt.startswith('-'):
+                rem_opts.append(opt)
+            elif opt.startswith('#'):
+                verb_opts.append(opt)
+            else:
+                select_opt += ':' + opt
+
+        return [self._format_option(select_opt),
+                *(self._format_option(opt) for opt in rem_opts),
+                *verb_opts]
+
+    def emit_preamble(self, job):
+        preamble = [
+            self._format_option('-N "%s"' % job.name),
+            self._format_option('-o %s' % job.stdout),
+            self._format_option('-e %s' % job.stderr),
+            self._format_option('-wd %s' % job.workdir),
+        ]
+
+        if job.time_limit is not None:
+            h, m, s = seconds_to_hms(job.time_limit)
+            preamble.append(
+                self._format_option('-l h_rt=%d:%d:%d' % (h, m, s)))
+
+        preamble += self._emit_lselect_option(job)
+
+        return preamble
+
+    def submit(self, job):
+        # `-o` and `-e` options are only recognized in command line by the PBS,
+        # SGE, and Slurm wrappers.
+        cmd = f'qsub -o {job.stdout} -e {job.stderr} {job.script_filename}'
+        completed = _run_strict(cmd, timeout=self._submit_timeout)
+        jobid_match = re.search(r'^Your job (?P<jobid>\S+)', completed.stdout)
+        if not jobid_match:
+            raise JobSchedulerError('could not retrieve the job id '
+                                    'of the submitted job')
+
+        job._jobid = jobid_match.group('jobid')
+        job._submit_time = time.time()
+
+    def poll(self, *jobs):
+        if jobs:
+            # Filter out non-jobs
+            jobs = [job for job in jobs if job is not None]
+
+        if not jobs:
+            return
+
+        user = osext.osuser()
+        completed = osext.run_command(
+            f'qstat -xml -u {user}'
+        )
+
+        if completed.returncode != 0:
+            raise JobSchedulerError(
+                f'qstat failed with exit code {completed.returncode} '
+                f'(standard error follows):\n{completed.stderr}'
+            )
+
+        root = ET.fromstring(completed.stdout)
+
+        # Store information for each job separately
+        jobinfo = {}
+        for queue_info in root:
```

Comment:
Also from what I can see, I think this refers to the list of jobs, right? If yes, perhaps a job-related name would be more relevant.

Reply:
Is the second entry (that with the pending job) in this example indeed a `job_info`? I would expect it to be `queue_info` as well. Otherwise, the output is totally weird :-D